### ISSUE TITLE
(AppDir) Intermittent error with Route Groups breaking render

### ISSUE THREAD

**Issue by zackdotcomputer** [vercel/next.js#44510]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: linux
      Arch: x64
      Version: #36-Ubuntu SMP Mon Dec 5 19:31:08 UTC 2022
    Binaries:
      Node: 16.17.0
      npm: 8.19.1
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.1.1-canary.1
      eslint-config-next: 13.0.7
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue

https://codesandbox.io/s/zen-joliot-kuwb91

### To Reproduce

Place your root `page.tsx` into a `(group)` directory, such as you might thing was possible from [this documentation](https://beta.nextjs.org/docs/routing/defining-routes#example-creating-multiple-root-layouts)

Then, try to navigate to `/`

### Describe the Bug

Following the reproduction steps, you should be met with `Uncaught Error: The default export is not a React Component in page: "/"`

### Expected Behavior

My expectation would be one of the following:
1. That a naked `page.tsx` in a top-level `(group)` folder would behave as `/`. This would include causing a server error if multiple groups contained one.
2. OR that there would be a way to define the "root page" in `next.config`
3. OR that there would be a defined folder-name that one had to put `page.tsx` inside if you wanted it to live somewhere other than at the root. E.G. that one could put it in an `index` or `root` folder inside of a `(group)` to have it behave the same as a naked `app/page.tsx`.

### Which browser are you using? (if relevant)

Not relevant - tested in Chrome and Firefox

### How are you deploying your application? (if relevant)

Currently just `next dev`

**Comment by zackdotcomputer** [comment]
Is it possible that route groups as a feature are broken in `canary`? Trying to move a different path into a route group is also failing with the same error.

**Comment by zackdotcomputer** [comment]
Sorry for wasting your time - I cannot fathom why this was broken on my machine earlier but a nuking of the build cache and dependencies has fixed it... If I figure out what caused that, I will be back.

**Comment by zackdotcomputer** [comment]
I walked away from this project for a couple days and returned. When I started up `yarn dev` today without changing the code, I got this error again. `rm -rf .next` and restarting did not fix it. Nor did removing `node_modules`. Instead locally I'm consistently getting:
```
Error: The default export is not a React Component in page: "/"
    at createComponentTree (~/code/app-name/node_modules/next/dist/server/app-render.js:759:27)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ~/code/app-name/node_modules/next/dist/server/app-render.js:819:56
    at async Promise.all (index 0)
    at async createComponentTree (~/code/app-name/node_modules/next/dist/server/app-render.js:783:38)
    at async ~/code/app-name/node_modules/next/dist/server/app-render.js:1059:51 {
  digest: '980462726'
```

However, my code sandbox is still not reproing this issue... Reopening it while I investigate.

**Comment by zackdotcomputer** [comment]
Figured out the issue and updated the CodeSandbox.

TLDR; If you have a nested layout that has a render error, e.g. forgets to make the layout a `default` export in my case, then the error message that renders says the problem is with the page and not with the problematic layout.
